### DISCUSSION TITLE
Homespan Thermostat with DHT22 sensor and relay

### DISCUSSION THREAD

**Discussion by plangenh** [HomeSpan/HomeSpan#396] [General]
First of all, thanks a lot for this wonderful library! It is truly awesome that we can integrate projects into HomeKit with a simple ESP32!

I am working on a thermostat project that combines a DHT22 and a relay to switch the heating mat of my Formicarium on and off.
This could be interesting for anyone with a terrarium as well…
 
I am successfully running the DHT 22 as temperature sensor and humidity sensor in homekit, but can't figure a way to set up a thermostat accessory category into the code with a relay included. 

Any help is much appreciated!!
attached is my current code

[HomeSpan-Thermostat.zip](https://github.com/HomeSpan/HomeSpan/files/9636527/HomeSpan-Thermostat.zip)

**Comment by HomeSpan** [comment]
To create a thermostat, you actually want to use the HomeKit Thermostat Service.   This has Characteristics for temperature and humidity - you don't need to create separate Services for a Temperature Sensor or Humidity Sensor (these are only used for standalone sensors).  See the HAP-R2 guide for the details on the Thermostat Service.
